testsuite/pom.xml

```diff
@@ -61,6 +53,41 @@
 
     <profiles>
         <profile>
+            <id>default</id>
+            <activation>
+                <property>
+                    <name>!forward.compatibility</name>
+                </property>
+            </activation>
+            <modules>
+                <module>arquillian-utils</module>
+                <module>unit-tests</module>
+                <module>legacy-integration-tests</module>
+                <module>integration-tests</module>
+                <module>integration-tests-spring</module>
+            </modules>
+        </profile>
+        <!-- Forward compatibility: use EAP 7.0.0.GA and newer Client from CP stream (e.g. client from RESTEasy delivered by EAP 7.0.2 -->
+        <profile>
+            <id>forward.compatibility</id>
+            <activation>
+                <property>
+                    <name>forward.compatibility</name>
+                </property>
+            </activation>
+            <modules>
+                <module>arquillian-utils</module>
+                <module>unit-tests</module>
+                <!-- Uncomment module below when EAP/Widlfly with Resteasy 3.1 is released -->
```

Comment:
When EAP/Wildfly is released with 3.1.0.Final, we will use this newly released container as "old" version to run against new versions of client and testsuite. Now the forward compatibility tests are run against EAP 7.0.0.GA which doesn't have resteasy-legacy module so the tests cannot pass. So, no, this PR is not required only until Resteasy Final. 
Please correct me, if it doesn't make sense to you.